compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -16,18 +31,23 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
 ) -> Result<(), ErrorHandled> {
     debug!("is_const_evaluatable({:?}, {:?})", def, substs);
     if infcx.tcx.features().const_evaluatable_checked {
-        // FIXME(const_evaluatable_checked): Actually look into generic constants to
-        // implement const equality.
-        for pred in param_env.caller_bounds() {
-            match pred.skip_binders() {
-                ty::PredicateAtom::ConstEvaluatable(b_def, b_substs) => {
-                    debug!("is_const_evaluatable: caller_bound={:?}, {:?}", b_def, b_substs);
-                    if b_def == def && b_substs == substs {
-                        debug!("is_const_evaluatable: caller_bound ~~> ok");
-                        return Ok(());
+        if let Some(ct) = AbstractConst::new(infcx.tcx, def, substs) {
```

Comment:
We only need to build this in case we ever hit the situation where `b_def != def || b_substs != substs`. I understand that this only computes it once even if needed multiple times, but maybe it can be cached in a mutable local `Option`?

Reply:
Oh, this is super cheap for !polymorphic const bodies... carry on